### PR TITLE
The Canadian Issue

### PR DESCRIPTION
**Pull request by markjamesbutler**
# The Canadian issue
## What does this change?

Fixes: Canada appears to be missing from the phone number drop-down on (https://profile.theguardian.com/account/edit)
## What is the value of this and can you measure success?

UserHelp don't receive reports of errors that Canada is missing from the drop down.  It additionally, fixes a bug involving selecting the wrong country when initially loading page.  For example, if the user has set a +1 country code, on page load the US Virgin Islands would be selected as default.  This is due to this country being the last selected item in the options, as there are multiple options with the value 1.

As the USA and Canada share country codes we cannot easily determine which country the user is located base on their phone.  Unfortunately the API implementation of this issue does not save the country with the telephone number thus allowing us to select the correct country from the drop down.

Please note a longer term fix for this issue is to store the country against the phone number so this can be shown in the dialog.
## Does this affect other platforms - Amp, Apps, etc?

No
## Screenshots

screen shot 2016-04-05 at 16 53 19
## 

_Does this PR meet the [contributing guidelines](https://github.com/guardian/frontend/blob/issue_pr_templates/.github/CONTRIBUTING.md#submission)?_

### CODE REVIEW

File: identity/app/model/CountryCodes.scala [hunk 1]
```diff
@@ -2,242 +2,14 @@ package model
 
 object CountryCodes {
```
Comment (kelvin-chappell) [inline identity/app/model/CountryCodes.scala]: Perhaps this should be called something like `PhoneNumbers` to make it clear what the codes are?

Comment (markjamesbutler) [inline identity/app/model/CountryCodes.scala]: Good suggestion. :)

**Comment by emma-p** [comment]
Should there be a label explaining what the dropdown is for? I don't think it is obvious if you remove the country name

**Comment by markjamesbutler** [comment]
@emma-p How about this as a pragmatic fix.  We should iterate on this in later releases as the UX around phone number entry is quite complicated.

screen shot 2016-04-12 at 18 23 26

**Comment by emma-p** [comment]
:+1: That looks good

File: identity/app/model/PhoneNumbers.scala [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package model
+
+object PhoneNumbers {
+
+  lazy val countryCodes = List(1, 7, 20, 27, 30, 31, 32, 33, 34, 36, 39, 40, 41, 43, 44, 45, 46, 47, 48, 49, 51, 52, 53,
```
Comment (jamespamplin) [inline identity/app/model/PhoneNumbers.scala]: This doesn't look very maintainable, what happens if you've left out another country or its code? Would a Map of Country to Country Code be more maintainable?

File: identity/app/model/PhoneNumbers.scala [hunk 3]
```diff
@@ -1,15 +1,23 @@
...
+
+  /**
+    * Unfortunately the country code map is not accessible in PhoneNumberUtil. This code uses reflection to obtain access
+    * then transform it to the correct structure.
+    */
+  lazy val countryCodes = {
+    val m = ru.runtimeMirror(phoneNumberUtil.getClass.getClassLoader)
+    val countryCallingCodeToRegionCodeMapSymb = ru.typeOf[PhoneNumberUtil].decl(ru.TermName("countryCallingCodeToRegionCodeMap")).asTerm
+    val im = m.reflect(phoneNumberUtil)
+    val countryCallingCodeToRegionCodeMap = im.reflectField(countryCallingCodeToRegionCodeMapSymb).get.asInstanceOf[java.util.HashMap[java.lang.Integer,java.util.List[java.lang.String]]]
```
Comment (jamespamplin) [inline identity/app/model/PhoneNumbers.scala]: Crazy code, but still better than hardcoding ourselves.

**Comment by jamespamplin** [comment]
:+1: good stuff

**Pull request merged**